packages/ketcher-core/src/domain/serializers/ket/ketSerializer.ts

```diff
@@ -715,10 +715,15 @@ export class KetSerializer implements Serializer<Struct> {
 
         if (monomer instanceof AmbiguousMonomer) {
           templateId =
-            monomer.variantMonomerItem.id ||
-            monomer.monomers.reduce(
-              (templateId, monomer) =>
-                templateId + '_' + getMonomerUniqueKey(monomer.monomerItem),
+            monomer.variantMonomerItem.subtype +
+            '_' +
+            monomer.variantMonomerItem.options.reduce(
+              (templateId, option) =>
+                templateId +
+                '_' +
+                option.templateId +
+                '_' +
+                (option.probability || option.ratio || ''),
```

Comment:
If both probability and ratio are not defined, we will end up with string having underscore in the end: 'TemplateId_TemplateId_'. I'm not sure if this case is possible and if so, will it affect anything, just wanted to highlight

Reply:
Thanks, it should not affect I think